Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2144,16 +2144,6 @@ extension BrowserViewController: LegacyTabDelegate {
 
 // MARK: - LibraryPanelDelegate
 extension BrowserViewController: LibraryPanelDelegate {
```

Comment:
We shouldn't modify BrowserViewController exiting code. The library coordinator will not exists for all code path and we need to make sure the current refactor has no effect on production code. If we need a different way of doing for coordinator, then we should protect the code change with feature flags.

This applies to a lot of the changes I see in this PR. The PR is changing a lot of production code, which goes beyond the scope of the work.